components/function-controller/internal/controllers/serverless/function_reconcile.go

```diff
@@ -58,7 +58,7 @@ func (r *FunctionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Owns(&autoscalingv1.HorizontalPodAutoscaler{}).
 		WithOptions(controller.Options{
-			MaxConcurrentReconciles: r.config.MaxConcurrentReconciles,
+			MaxConcurrentReconciles: 1, // Build job scheduling mechanism requires this parameter to be set to 1 https://github.com/kyma-project/kyma/issues/10037
```

Comment:
After some time, adding the link is not good thing, because the issue itself doesn't say why the value is 1.
It only describe that we want to implement concurrent build jobs limit.

I would add description why the value is 1. How it affects the scheduling mechanism. Why will happen when the value is set to something different bigger than 1.